src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
You know, the more I play with this, the more I think I'm on the wrong path. `Apply` and `Applicative` are both generic traits over higher kinded types: `Apply[F[_]]`. I can't get the types to work right in Java since you can't do this. Do you think that you'll have any other `Applicative` types in Javaslang? Because if not, then it might make sense to abandon my approach with `Apply` and `Applicative` and just keep all of this behavior in `Validation`. What are your thoughts?